src/mlpack/methods/cf/cf.hpp

```diff
@@ -258,22 +258,24 @@ class CF
   //! Cleaned data matrix.
   arma::sp_mat cleanedData;
 
-  /**
-   * Helper function to insert a point into the recommendation matrices.
-   *
-   * @param queryIndex Index of point whose recommendations we are inserting
-   *     into.
-   * @param pos Position in list to insert into.
-   * @param neighbor Index of item being inserted as a recommendation.
-   * @param value Value of recommendation.
-   */
-  void InsertNeighbor(const size_t queryIndex,
-                      const size_t pos,
-                      const size_t neighbor,
-                      const double value,
-                      arma::Mat<size_t>& recommendations,
-                      arma::mat& values) const;
-
+  //! Candidate represents a possible recommendation.
+  struct Candidate
+  {
+    //! Value of this recommendation.
+    double value;
+    //! Item of this recommendation.
+    size_t item;
+    //! Trivial constructor.
+    Candidate(double value, size_t item) :
+        value(value),
+        item(item)
+    {};
+    //! Compare the value of two candidates.
+    friend bool operator>(const Candidate& l, const Candidate& r)
+    {
+      return l.value > r.value;
+    };
+  };
```

Comment:
@rcurtin I think you see a difference in runtime between std::pair and SortStruct in: #712  , because you use a pointer to function comparison parameter instead of a functor or an operator, as I can see in: https://github.com/mlpack/mlpack/pull/721/commits/640f3a7859ea8d400bc6fed6b4395c2291ba0259#diff-3a0c62f44f497b9857aed01fe81dc391L54
I have read some documentation that says that functors can be inlined while pointer to function generally are not.
For Candidate struct, I define an operator, that probably will be inlined by the compiler.